### ISSUE TITLE
Can't delcare variable type by comment

### ISSUE THREAD

**Issue by pawelkg** [vimeo/psalm#2178]
This example shouldn't show error: https://psalm.dev/r/f7478fba75
Psalm should respect variable declaration via @var tag.

**Comment by andrew-demb** [comment]
Use `/** @var myClass $myClass */` instead `/* @var myClass $myClass */`

**Comment by pawelkg** [comment]
In my version of Psalm it's still showing errors. My version is: Psalm 3.5.2@648dda67ed7aca1d4ea243258f0ee7f2844ac60a

**Comment by andrew-demb** [comment]
I added new `*` and psalm now doesn't show any error.
https://psalm.dev/r/82e7755db4

> Psalm output (using commit 4823ec5):
No issues!

**Comment by muglug** [comment]
Yeah `/*` isn’t a valid docblock

**Comment by pawelkg** [comment]
OK. I fix it to /** but I still have error.
My code is:
```
$restrictionRelationship = $this->restriction
            ->getAttribute('relationship');

        if (null !== $restrictionRelationship) {
            /** @var \Wszetko\Sitemap\Items\DataTypes\StringType $restrictionRelationship */
            $restrictionRelationship
                ->setConversion('lower')
                ->setAllowedValues('allow, deny');
        }
```
getAttribute() method have return type is AbstractDataType but in this case it's return StringType (like in docblock; StringType extends AbstractDataType). And my result is:
```
ERROR: UndefinedMethod - src\Items\Video.php:323:19 - Method Wszetko\Sitemap\Items\DataTypes\AbstractDataType::setconversion does not exist
                ->setConversion('lower')
```

And by the way, it's change in message my method name to lowercase (setconversion) and it should be 'setConversion'.

**Comment by muglug** [comment]
OH currently `@var` doesn't work above method calls (ticketed in #1916).

You can fix in your example by putting the `@var` above the assignment:

```php
/** @var ?\Wszetko\Sitemap\Items\DataTypes\StringType $restrictionRelationship */
$restrictionRelationship = $this->restriction
            ->getAttribute('relationship');
```
